src/main/java/org/jvnet/hudson/plugins/thinbackup/ThinBackupPluginImpl.java

```diff
@@ -61,17 +58,21 @@ public class ThinBackupPluginImpl extends Plugin {
     private boolean failFast = true;
 
     @Override
-    public void start() throws Exception {
-        super.start();
+    public boolean configure(StaplerRequest request, JSONObject jsonObject) {
+        return true;
+    }
+
+    public ThinBackupPluginImpl() {
         load();
         LOGGER.fine("'thinBackup' plugin initialized.");
     }
 
-    public static ThinBackupPluginImpl getInstance() {
-        final Jenkins jenkins = Jenkins.getInstanceOrNull();
-        if (jenkins != null) {
-            return jenkins.getPlugin(ThinBackupPluginImpl.class);
-        } else {
+    public static ThinBackupPluginImpl get() {
+        try {
+            ExtensionList<GlobalConfiguration> all = GlobalConfiguration.all();
+            return all.get(ThinBackupPluginImpl.class);
+        } catch (IllegalStateException exception) {
+            // probably no Jenkins instance running
             return null;
         }
```

Comment:
In the latest [archetype](https://github.com/jenkinsci/archetypes/blob/e92ac3efd553e4e96a2ff8eb51b15cf9301191a2/global-configuration/src/main/resources/archetype-resources/src/main/java/SampleConfiguration.java) we recommend this pattern instead:

```suggestion
        return ExtensionList.lookupSingleton(ThinBackupPluginImpl.class);
```